mempool/clist_mempool.go

```diff
@@ -850,21 +852,25 @@ func (iter *CListIterator) WaitNextCh() <-chan Entry {
 		if iter.cursor == nil {
 			// We are at the beginning of the iteration or the saved entry got removed: wait until
 			// the list becomes not empty and select the first entry.
-			<-iter.txs.WaitChan()
+			select {
+			case <-iter.txs.WaitChan():
+			case <-iter.ctx.Done():
```

Comment:
Do you think we shouldn't exit here?

```suggestion
			case <-iter.ctx.Done():
			   close(ch)
			   return
```